packages/@o3r/core/schematics/component/common/context.ts

```diff
@@ -0,0 +1,77 @@
+import { externalSchematic, Rule, SchematicContext } from '@angular-devkit/schematics';
+import { askConfirmation, askQuestion } from '@angular/cli/src/utilities/prompt';
+import { setupSchematicsDefaultParams } from '@o3r/schematics';
+import { NgGenerateComponentSchematicsSchema } from '../schema';
+
+const contextPackageName = '@o3r/core';
+const addContextSchematicName = 'context-to-component';
+
+export const getAddContextRules = async (
+  componentPath: string,
+  options: Pick<NgGenerateComponentSchematicsSchema, 'useContext' | 'skipLinter'>,
+  context: SchematicContext
+): Promise<Rule[]> => {
+  let useContext = options.useContext;
+  let alwaysUseContext;
+  try {
+    require.resolve(`${contextPackageName}/package.json`);
+    if (useContext === null && context.interactive) {
+      useContext = await askConfirmation('Generate component with Otter context ?', true);
+      if (useContext) {
+        alwaysUseContext = await askQuestion('Generate future components with Otter context by default?', [
+          {
+            type: 'choice',
+            name: 'Yes, always',
+            value: 'yes'
+          },
+          {
+            type: 'choice',
+            name: 'Ask me again next time',
+            value: 'ask-again'
+          },
+          {
+            type: 'choice',
+            name: 'No, don\'t apply Otter context by default',
+            value: 'no'
+          }
+        ], 0, null);
+      } else {
+        context.logger.info(`
+          You can add context later to this component via this command:
+          ng g ${contextPackageName}:${addContextSchematicName} --path="${componentPath}"
+        `);
+      }
+    }
+  } catch {
```

Comment:
I think you can remove the try/catch. the script cannot be run if `o3r/core` is not already installed.